internal/p2p/uploads.go

```diff
@@ -8,136 +8,198 @@ import (
-						}
-
-						// fmt.Println("[INFOSEC] File Security check passed for file: " + f + " with SHA256: " + fsha256)
-						fresult = result
-					}
-
-				}
-			}
-
-			if fresult {
-				fcid := psfs.AddFileIPFS(folderPath + "/" + f)
-				if fcid != "" {
-					fmt.Println("[INFO] File: " + f + " ++added to IPFS with CID: " + fcid)
-					fileExtension, err := psfs.GetExtension(f)
-					if err != nil {
-						return
-					}
-
-					metadata := store.BaseMetadata{
-						FileSHA256: strings.ToLower(fsha256),
-						IPFSCID:    strings.ToLower(fcid),
-						FileType:   strings.ToLower(fileExtension),
-					}
-
-					errgs := store.GlobalStore.AddFile(metadata)
-					if errgs != nil {
-						fmt.Printf("[ERROR] failed to add file to GlobalStore: %w \n", errgs)
-						return
-					}
-					fmt.Println("[INFO] File: " + f + " ++added to GlobalStore with CID: " + fcid)
-					count = count + 1
-					if appconfInstance.FFMoveUpload {
-						err := psfs.MoveFile(folderPath+"/"+f, appconfInstance.CacheFolder+"/"+f)
-						if err != nil {
-							fmt.Println("[ERROR] Error moving file: ", err)
-						}
-					}
-				}
-			} else {
-				if appconfInstance.FFSendFileVT {
-					// This was really to catch unknow files on VT
-
-					// TODO: 				if appconfInstance.SecurityCapability [1 2 3 4]
-
-					if appconfInstance.SecurityCapability == 4 {
-						fmt.Println("[INFO] Submitting File to 3rd Party for Security check for file: " + f + " with SHA256: " + fsha256)
-						submitresult, err := psfs.SendFileToVirusTotalWS(folderPath + "/" + f)
-						if err != nil {
-							fmt.Println("[ERROR] Error submitting file for security check: ", err)
-						}
-						if submitresult {
-							fmt.Println("[INFO] Submission Passed Security check for file: " + f + " with SHA256: " + fsha256)
-						} else {
-							fmt.Println("[ERROR] File Security check failed for file: " + f + " with SHA256: " + fsha256)
-							if appconfInstance.FFMoveUpload {
-								err := psfs.MoveFile(folderPath+"/"+f, appconfInstance.RejectFolder+"/"+f)
-								if err != nil {
-									fmt.Println("[ERROR] Error moving file: ", err)
-								}
-							}
-						}
-					}
-				} else {
-					fmt.Println("[ERROR] File Security check failed for file: " + f + " with SHA256: " + fsha256)
-				}
-			}
-		} else {
-			fmt.Println("[ERROR] File type invalid for file: " + f)
-			if appconfInstance.FFMoveUpload {
-				err := psfs.MoveFile(folderPath+"/"+f, appconfInstance.RejectFolder+"/"+f)
-				if err != nil {
-					fmt.Println("[ERROR] Error moving file: ", err)
-				}
-			}
-			// move to rejected folder
-			// log reason in rejected folder logfile
+
+	var count int
+	for _, f := range files {
+		if processFile(folderPath, f) {
+			count++
 		}
 	}
+
 	if count >= 1 {
 		store.GlobalStore.Save(appconfInstance.MetaDataFile)
 	}
 }
+
+// processFile handles a single file upload. Returns true if the file was successfully added.
+func processFile(folderPath, f string) bool {
+	filePath := folderPath + "/" + f
```

Comment:
`"/"` isn't very OS-agnostic. Let's use `path.Separator` or something here and/or other stdlib path building apis.